### PR TITLE
DfTx: `withdrawFromVault`

### PR DESCRIPTION
**Pull request by ivan-zynesis**
#### What kind of PR is this?:

/kind feature

#### Which issue(s) does this PR fixes?:

Fixes #

### CODE REVIEW

File: packages/jellyfish-transaction/src/script/dftx/dftx_loans.ts [hunk 1]
```diff
@@ -263,6 +263,32 @@ export class CDepositToVault extends ComposableBuffer<DepositToVault> {
   }
 }
 
+/**
+ * WithdrawFromVault DeFi Transaction
+ */
+export interface WithdrawFromVault {
+  vaultId: string // ------------------| 32 bytes, Vault Id
+  to: Script // ---------------------| n = VarUInt{1-9 bytes}, + n bytes, Address to receive withdrawn collateral
```
Comment (canonbrother) [inline packages/jellyfish-transaction/src/script/dftx/dftx_loans.ts:271]: alignment

**Review by canonbrother — APPROVED** [review]

File: packages/jellyfish-transaction-builder/__tests__/txn/txn_builder_loan_withdraw_from_vault.test.ts [hunk 2]
```diff
@@ -0,0 +1,372 @@
...
+      const destination = await tGroup.get(0).generateAddress()
+      const decoded = fromAddress(destination, 'regtest')
+      const script = decoded?.script as Script
+
+      // set oracle price above spent the utxos and change go into b58 address
+      // causing no prevout at address derived directly from elliptic pair, fund for utxo again
+      await fundEllipticPair(tGroup.get(0).container, providers.ellipticPair, 10)
+
+      const what = await tGroup.get(0).rpc.loan.getVault(dualCollateralVault)
+      console.log(what)
```
Comment (surangap) [inline packages/jellyfish-transaction-builder/__tests__/txn/txn_builder_loan_withdraw_from_vault.test.ts:325]: remove what?😅

File: packages/jellyfish-transaction-builder/__tests__/txn/txn_builder_loan_withdraw_from_vault.test.ts [hunk 3]
```diff
@@ -0,0 +1,372 @@
...
+        vaultId: dualCollateralVault,
+        to: script,
+        tokenAmount: { token: 0, amount: new BigNumber(6000) } // DFI (USD value) remaining will become less than 50%
+      }, vaultOwnerScript)
+
+      const promise = sendTransaction(tGroup.get(0).container, txn)
+      // await promise
+      // await tGroup.get(0).generate(1)
+      // const withdrawn = await tGroup.get(0).rpc.loan.getVault(vaultId1)
+      // console.log('should not', withdrawn)
```
Comment (surangap) [inline packages/jellyfish-transaction-builder/__tests__/txn/txn_builder_loan_withdraw_from_vault.test.ts:337]:
```suggestion
```

**Review by surangap — COMMENTED** [review]
looks good. minor comments

File: packages/jellyfish-transaction-builder/__tests__/txn/txn_builder_loan_withdraw_from_vault.test.ts (hunk 2 above)
Comment (ivan-zynesis) [reply packages/jellyfish-transaction-builder/__tests__/txn/txn_builder_loan_withdraw_from_vault.test.ts:325]: oops, the bug was on defid, this test wasn't able to pass. Forgotten them after too long

File: packages/jellyfish-transaction/src/script/dftx/dftx_loans.ts (hunk 1 above)
Comment (ivan-zynesis) [reply packages/jellyfish-transaction/src/script/dftx/dftx_loans.ts]: oops, they never align on my editor because some of my font

**Review by surangap — APPROVED** [review]

**Review by fuxingloh — APPROVED** [review]
We need to add more frozen state tests, not sure how well covered it is in our test suit.

**Pull request merged**